src/snap/metric.cc

```diff
@@ -190,6 +188,26 @@ void Metric::set_data(int data) {
   rpc_metric_ptr->set_int32_data(data);
 }
 
+void Metric::set_data(long int data) {
+  type = DataType::Int64;
```

Comment:
if compiling in 32 bit mode, long would only be int32--I would suggest updating to be explicit and use int32_t, int64_t, uint32_t, uint64_t as well.